wallet/wallet.go

```diff
@@ -1235,15 +1256,25 @@ out:
 // transaction creation through this function is serialized to prevent the
 // creation of many transactions which spend the same outputs.
 //
+// A set of functional options can be passed in to apply modifications to the
+// tx creation process such as using a custom change scope, which otherwise
+// defaults to the same as the specified coin selection scope.
+//
 // NOTE: The dryRun argument can be set true to create a tx that doesn't alter
-// the database. A tx created with this set to true SHOULD NOT be broadcasted.
-func (w *Wallet) CreateSimpleTx(keyScope *waddrmgr.KeyScope, account uint32,
-	outputs []*wire.TxOut, minconf int32, satPerKb btcutil.Amount,
-	coinSelectionStrategy CoinSelectionStrategy, dryRun bool) (
-	*txauthor.AuthoredTx, error) {
+// the database. A tx created with this set to true SHOULD NOT be broadcast.
+func (w *Wallet) CreateSimpleTx(coinSelectKeyScope *waddrmgr.KeyScope,
+	account uint32, outputs []*wire.TxOut, minconf int32,
+	satPerKb btcutil.Amount, coinSelectionStrategy CoinSelectionStrategy,
+	dryRun bool, optFuncs ...TxCreateOption) (*txauthor.AuthoredTx, error) {
```

Comment:
Makes sense.  Users of `SendOutputs`, like myself, would benefit from the new are too though.
It looks simple enough to adapt and instead do `CreateSimpleTx` directly, followed by `PublishTransaction`... just noting.
Trying this PR out with a `replace` now to see.  Thanks for adapting!

Reply:
Seems I can write `SendOutputs` ourselves like the following, which is close, but not quite the same.

```go

func (w *spvWallet) sendOutputs(outputs []*wire.TxOut, keyScope *waddrmgr.KeyScope, account uint32, minconf int32,
	satPerKb btcutil.Amount, coinSelectionStrategy wallet.CoinSelectionStrategy, label string) (*wire.MsgTx, error) {
	// Ensure the outputs to be created adhere to the network's consensus rules.
	for _, output := range outputs {
		err := txrules.CheckOutput(output, txrules.DefaultRelayFeePerKb)
		if err != nil {
			return nil, err
		}
	}

	// Create the transaction and broadcast it to the network. The
	// transaction will be added to the database in order to ensure that we
	// continue to re-broadcast the transaction upon restarts until it has
	// been confirmed.
	changeTypeOpt := wallet.WithCustomChangeScope(&waddrmgr.KeyScopeBIP0084)
	createdTx, err := w.wallet.CreateSimpleTx(keyScope, account, outputs, minconf,
		satPerKb, coinSelectionStrategy, false, changeTypeOpt)
	if err != nil {
		return nil, err
	}

	err = w.wallet.PublishTransaction(createdTx.Tx, label)
	if err != nil {
		return nil, err
	}

	return createdTx.Tx, nil
}
```

Reply:
It's also worth noting that although the new signature for `CreateSimpleTx`, is close and the final arg is optional, it is still a breaking change because `interface` definitions will no longer match.

That's not terribly hard to adapt to, but it is something to consider about why a method with a different name could be better.
If a project uses btcwallet directly, plus it uses another project that also uses btcwallet and this method, you cannot update btcwallet until that _other project_ you required updates (all depending on if/how the projects define interfaces that `btcwallet.Wallet` implements).

e.g.

```go
type BTCWallet interface {
	PublishTransaction(tx *wire.MsgTx, label string) error
	ListUnspent(minconf, maxconf int32, acctName string) ([]*btcjson.ListUnspentResult, error)
	... etc
	Unlock(passphrase []byte, lock <-chan time.Time) error
	SendOutputs(outputs []*wire.TxOut, keyScope *waddrmgr.KeyScope, account uint32, minconf int32,
		satPerKb btcutil.Amount, coinSelectionStrategy wallet.CoinSelectionStrategy, label string) (*wire.MsgTx, error)
	// **** ERROR: "does not implement" - have / want" at compile:
	CreateSimpleTx(coinSelectKeyScope *waddrmgr.KeyScope, account uint32,
		outputs []*wire.TxOut, minconf int32, satPerKb btcutil.Amount,
		coinSelectionStrategy wallet.CoinSelectionStrategy, dryRun bool,
		optFuncs ...wallet.TxCreateOption) (*txauthor.AuthoredTx, error)
}
```

Reply:
> It looks simple enough to adapt and instead do CreateSimpleTx directly, followed by PublishTransaction... just noting.

The functional difference being that `SendOutputs` will publish for you? Or is it the extra tx rules checks?

Reply:
Ah ok you'd need to apply that label manually when broadcasting, but yeah that looks functionally the same.

Reply:
Yeah, the label, and I suppose the other things are that we don't get the txhash check when we use `PublishTransaction` because it eats the txhash output.  There's also a watch only wallet check, but that doesn't really affect use I think.
NBD, just not an exact clone of `SendOutputs`